packages/liquidator/src/walletBalanceAlarm.js

```diff
@@ -0,0 +1,157 @@
+      const targetUsdcWalletBalance = await this.calculateUsdcTargetBalance();
+      console.log("checkDisputerBotBalanceAgainstStrategy called...");
+      console.log("Current Collateral Balance: ", currentCollateralBalance);
+      console.log("Target USDC Wallet Balance: ", targetUsdcWalletBalance);
+
+      if (currentCollateralBalance < targetUsdcWalletBalance) {
+        this.logger.warn({
+          at: "Disputer#WalletBalanceAlarm",
+          message: `Disputer bot wallet balance is ${currentCollateralBalance.toString()} which is below the target USDC wallet balance threshold of ${targetUsdcWalletBalance.toString()}. Replenish disputer bot wallet balance immediately`,
+          numOfOpenPositions: `${this.numOfOpenPositions}`,
+          totalCollateralAmount: `${this.totalCollateralAmount}`,
+          disputeBondPercentage: `${this.disputerBondPercentage}`,
+          currentCollateralBalance: `${currentCollateralBalance}`,
+          targetUsdcWalletBalance: `${targetUsdcWalletBalance}`,
+        });
+        return true;
+      } else {
+        this.logger.info({
+          at: "Disputer#WalletBalanceAlarm",
+          message: `Current disputer bot wallet balance of ${currentCollateralBalance.toString()} meets the target USDC wallet balance threshold of ${targetUsdcWalletBalance.toString()}. Disputer bot wallet USDC balance is within the healthy range.`,
+          numOfOpenPositions: `${this.numOfOpenPositions}`,
+          totalCollateralAmount: `${this.totalCollateralAmount}`,
+          disputeBondPercentage: `${this.disputerBondPercentage / 1e18}`,
+          currentCollateralBalance: `${currentCollateralBalance}`,
+          targetWalletBalance: `${targetUsdcWalletBalance}`,
+        });
+        return false;
+      }
+    } catch (error) {
+      this.logger.error({
+        at: "Disputer#WalletBalanceAlarm",
+        message: "An error occurred during the calculation of the disputer bot wallet balance",
+        error: error,
+        stack: error.stack,
+      });
+
+      throw error;
+    }
+  }
+
+  async calculateUsdcTargetBalance() {
+    const allPositions = await this.financialContractClient.getAllPositions();
+    this.totalCollateralAmount = await allPositions.reduce((total, position) => {
+      return total + parseFloat(position.collateral.toString());
+    }, 0);
+    console.log("calculateUsdcTargetBalance called...");
+    console.log(`Total Collateral Amount: ${this.totalCollateralAmount}`);
+
+    const ooReward = await this.financialContract.methods.ooReward().call();
+    console.log(`OO Reward: ${ooReward}`);
+
+    this.numOfOpenPositions = await this.financialContractClient.getAllPositions().length;
+    console.log(`Number of Open Positions: ${this.numOfOpenPositions}`);
+
+    this.disputerBondPercentage = await this.financialContract.methods.disputeBondPercentage().call();
+    console.log(`Dispute Bond Percentage: ${this.disputerBondPercentage / 1e18}`);
+
+    const usdcTargetBalance =
```

Comment:
This looks correct ✅

Reply:
I would just add extra brackets, to make sure correct order of calculation is being followed.

(this.totalCollateralAmount * (this.disputerBondPercentage / 1e18)) + ((this.numOfOpenPositions * this.bufferPercentage) * ooReward);